src/components/composer/grid_composer/grid_composer.ts

```diff
@@ -74,6 +89,39 @@ export class GridComposer extends Component<Props, SpreadsheetChildEnv> {
     });
   }
 
+  get shouldDisplayCellReference(): boolean {
+    if (this.isCellReferenceVisible) {
+      return true;
+    }
+    const sheetId = this.env.model.getters.getActiveSheetId();
+    const zone = this.env.model.getters.getSelectedZone();
+    const rect = this.env.model.getters.getVisibleRect(zone);
+    if (
+      !deepEquals(rect, this.rect) ||
+      sheetId !== this.env.model.getters.getCurrentEditedCell().sheetId
+    ) {
+      this.isCellReferenceVisible = true;
```

Comment:
kinda strange to have a getter writing stuff. Do we need to store the result? Is it to keep it always visible as soon as it appeared once? If yes, what do you think of doing this in a `willUpdateProps` ? That way it would be more obvious it's updating the internal state (which I don't expect from a getter)